internal/backup_mark.go

```diff
@@ -0,0 +1,50 @@
+package internal
+
+import (
+	"bytes"
+	"encoding/json"
+
+	"github.com/wal-g/wal-g/internal/storages/storage"
+	"github.com/wal-g/wal-g/utility"
+)
+
+// GetImpermanentBackupsBefore marks all previous related backups permanent,
+// including itself, any previous delta backups and initial full backup
+// TODO: unit tests
+func GetImpermanentBackupsBefore(baseBackupFolder storage.Folder, backupName string, toUpload *[]UploadObject) error {
```

Comment:
toUpload -> impermanentBackupMetadatas
Let's also return this list instead of passing it by the pointer.